### PR TITLE
warn_config_absent: Check at least 1 config present

### PR DESCRIPTION
**Pull request by adhikasp**
Modify ``warn_config_absent()`` to check list of options that at least
1 should be present. Currently, ``aspects`` option can replace
``bears`` and either one of those must present in a section for it
to run.

Fixes https://github.com/coala/coala/issues/4563

### CODE REVIEW

File: coalib/settings/ConfigurationGathering.py [hunk 1]
```diff
@@ -135,27 +135,30 @@ def warn_nonexistent_targets(targets, sections, log_printer):
```
Comment (hemangsk) [inline coalib/settings/ConfigurationGathering.py]: file name is missing in the shortlog

Comment (hemangsk) [inline coalib/settings/ConfigurationGathering.py]: minor type in commit message  "either one of those must *be* present"

Comment (adhikasp) [inline coalib/settings/ConfigurationGathering.py]: About the shortlog, I think the current one is more descriptive for the change and it doesn't exceed the char limit.

**Comment by Nosferatul** [comment]
@adhikasp rebase, I think it's good to merge 👍

**Comment by adhikasp** [comment]
rebased

**Comment by jayvdb** [comment]
needs another rebase

File: coalib/settings/ConfigurationGathering.py [hunk 2]
```diff
@@ -135,27 +135,30 @@ def warn_nonexistent_targets(targets, sections, log_printer):
...
-    bears_config_absent = warn_config_absent(sections, 'bears', log_printer)
+    files_config_absent = warn_config_absent(sections, ['files'], log_printer)
+    bears_config_absent = warn_config_absent(sections, ['bears', 'aspects'],
+                                             log_printer)
     if files_config_absent or bears_config_absent:
         raise SystemExit(2)  # Invalid CLI options provided
 
 
-def warn_config_absent(sections, argument, log_printer):
+def warn_config_absent(sections, arguments, log_printer):
```
Comment (jayvdb) [inline coalib/settings/ConfigurationGathering.py]: this is a breaking change.
The parameter name cant change, and the function must accept and correctly handle either a str or list of str

File: coalib/settings/ConfigurationGathering.py (hunk 1 above)
Comment (jayvdb) [inline coalib/settings/ConfigurationGathering.py]: It doesnt give context.  and `exchangeable` is too big a word for this.  simplify.

**Comment by jayvdb** [comment]
reack 726674a

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**